# Importing data from yahoo finance

# 1. Getting historical-data [i.e.  Open/high/low/close] using yfinance 

In [1]:
import pandas as pd
import yfinance as yf

## getting historical data for a particular stock

In [2]:
# go to yahoo finance and find the symbol/ticker for the instrument
ticker = "TSLA"
yf.download(ticker)

[*********************100%***********************]  1 of 1 completed


Price        Adj Close       Close        High         Low        Open  \
Ticker            TSLA        TSLA        TSLA        TSLA        TSLA   
Date                                                                     
2010-06-29    1.592667    1.592667    1.666667    1.169333    1.266667   
2010-06-30    1.588667    1.588667    2.028000    1.553333    1.719333   
2010-07-01    1.464000    1.464000    1.728000    1.351333    1.666667   
2010-07-02    1.280000    1.280000    1.540000    1.247333    1.533333   
2010-07-06    1.074000    1.074000    1.333333    1.055333    1.333333   
...                ...         ...         ...         ...         ...   
2025-02-10  350.730011  350.730011  362.700012  350.510010  356.209991   
2025-02-11  328.500000  328.500000  349.369995  325.100006  345.799988   
2025-02-12  336.510010  336.510010  346.399994  329.119995  329.940002   
2025-02-13  355.940002  355.940002  358.690002  342.850006  345.000000   
2025-02-14  355.839996  355.839996  362.000000  347.500000  360.619995   

Price          Volume  
Ticker           TSLA  
Date                   
2010-06-29  281494500  
2010-06-30  257806500  
2010-07-01  123282000  
2010-07-02   77097000  
2010-07-06  103003500  
...               ...  
2025-02-10   77514900  
2025-02-11  118543400  
2025-02-12  105382700  
2025-02-13   89441500  
2025-02-14   68054700  

[3682 rows x 6 columns]

## getting historical data for a given date-range

In [3]:
ticker = "TSLA"
start = '2020-01-01'
end = '2025-01-01'
yf.download(ticker,start, end) 

[*********************100%***********************]  1 of 1 completed


Price        Adj Close       Close        High         Low        Open  \
Ticker            TSLA        TSLA        TSLA        TSLA        TSLA   
Date                                                                     
2020-01-02   28.684000   28.684000   28.713333   28.114000   28.299999   
2020-01-03   29.534000   29.534000   30.266666   29.128000   29.366667   
2020-01-06   30.102667   30.102667   30.104000   29.333332   29.364668   
2020-01-07   31.270666   31.270666   31.441999   30.224001   30.760000   
2020-01-08   32.809334   32.809334   33.232666   31.215334   31.580000   
...                ...         ...         ...         ...         ...   
2024-12-24  462.279999  462.279999  462.779999  435.140015  435.899994   
2024-12-26  454.130005  454.130005  465.329987  451.019989  465.160004   
2024-12-27  431.660004  431.660004  450.000000  426.500000  449.519989   
2024-12-30  417.410004  417.410004  427.000000  415.750000  419.399994   
2024-12-31  403.839996  403.839996  427.929993  402.540009  423.790009   

Price          Volume  
Ticker           TSLA  
Date                   
2020-01-02  142981500  
2020-01-03  266677500  
2020-01-06  151995000  
2020-01-07  268231500  
2020-01-08  467164500  
...               ...  
2024-12-24   59551800  
2024-12-26   76366400  
2024-12-27   82666800  
2024-12-30   64941000  
2024-12-31   76825100  

[1258 rows x 6 columns]

## getting historical data for multiple stocks

In [4]:
ticker = ["TSLA","GOOG"]
yf.download(ticker) 

[*********************100%***********************]  2 of 2 completed


Price        Adj Close                   Close                    High  \
Ticker            GOOG        TSLA        GOOG        TSLA        GOOG   
Date                                                                     
2004-08-19    2.490186         NaN    2.499133         NaN    2.591785   
2004-08-20    2.687981         NaN    2.697639         NaN    2.716817   
2004-08-23    2.715032         NaN    2.724787         NaN    2.826406   
2004-08-24    2.602609         NaN    2.611960         NaN    2.779581   
2004-08-25    2.630652         NaN    2.640104         NaN    2.689918   
...                ...         ...         ...         ...         ...   
2025-02-10  188.199997  350.730011  188.199997  350.730011  189.990005   
2025-02-11  187.070007  328.500000  187.070007  328.500000  188.800003   
2025-02-12  185.429993  336.510010  185.429993  336.510010  186.830002   
2025-02-13  187.880005  355.940002  187.880005  355.940002  187.990005   
2025-02-14  186.869995  355.839996  186.869995  355.839996  188.149994   

Price                          Low                    Open              \
Ticker            TSLA        GOOG        TSLA        GOOG        TSLA   
Date                                                                     
2004-08-19         NaN    2.390042         NaN    2.490664         NaN   
2004-08-20         NaN    2.503118         NaN    2.515820         NaN   
2004-08-23         NaN    2.716070         NaN    2.758411         NaN   
2004-08-24         NaN    2.579581         NaN    2.770615         NaN   
2004-08-25         NaN    2.587302         NaN    2.614201         NaN   
...                ...         ...         ...         ...         ...   
2025-02-10  362.700012  187.610001  350.510010  189.059998  356.209991   
2025-02-11  349.369995  186.080002  325.100006  186.835007  345.799988   
2025-02-12  346.399994  183.630005  329.119995  185.229996  329.940002   
2025-02-13  358.690002  184.880005  342.850006  185.929993  345.000000   
2025-02-14  362.000000  186.110001  347.500000  186.830002  360.619995   

Price          Volume               
Ticker           GOOG         TSLA  
Date                                
2004-08-19  897427216          NaN  
2004-08-20  458857488          NaN  
2004-08-23  366857939          NaN  
2004-08-24  306396159          NaN  
2004-08-25  184645512          NaN  
...               ...          ...  
2025-02-10   16606000   77514900.0  
2025-02-11   13028100  118543400.0  
2025-02-12   17632300  105382700.0  
2025-02-13   12729300   89441500.0  
2025-02-14   12673300   68054700.0  

[5157 rows x 12 columns]

## getting historical data for multiple stocks for a given date range

In [5]:
ticker = ["TSLA","GOOG"]
start = '2020-01-01'
end = '2025-01-01'
yf.download(ticker, start, end) 

[*********************100%***********************]  2 of 2 completed


Price        Adj Close                   Close                    High  \
Ticker            GOOG        TSLA        GOOG        TSLA        GOOG   
Date                                                                     
2020-01-02   68.123718   28.684000   68.368500   28.684000   68.406998   
2020-01-03   67.789421   29.534000   68.032997   29.534000   68.625000   
2020-01-06   69.460922   30.102667   69.710503   30.102667   69.824997   
2020-01-07   69.417580   31.270666   69.667000   31.270666   70.149498   
2020-01-08   69.964615   32.809334   70.216003   32.809334   70.579002   
...                ...         ...         ...         ...         ...   
2024-12-24  197.570007  462.279999  197.570007  462.279999  197.669998   
2024-12-26  197.100006  454.130005  197.100006  454.130005  198.160004   
2024-12-27  194.039993  431.660004  194.039993  431.660004  196.800003   
2024-12-30  192.690002  417.410004  192.690002  417.410004  193.779999   
2024-12-31  190.440002  403.839996  190.440002  403.839996  193.250000   

Price                          Low                    Open              \
Ticker            TSLA        GOOG        TSLA        GOOG        TSLA   
Date                                                                     
2020-01-02   28.713333   67.077499   28.114000   67.077499   28.299999   
2020-01-03   30.266666   67.277199   29.128000   67.392998   29.366667   
2020-01-06   30.104000   67.500000   29.333332   67.500000   29.364668   
2020-01-07   31.441999   69.518997   30.224001   69.897003   30.760000   
2020-01-08   33.232666   69.542000   31.215334   69.603996   31.580000   
...                ...         ...         ...         ...         ...   
2024-12-24  462.779999  195.197998  435.140015  196.169998  435.899994   
2024-12-26  465.329987  195.869995  451.019989  196.740005  465.160004   
2024-12-27  450.000000  191.972000  426.500000  196.470001  449.519989   
2024-12-30  427.000000  190.360001  415.750000  190.865005  419.399994   
2024-12-31  427.929993  189.580002  402.540009  192.445007  423.790009   

Price         Volume             
Ticker          GOOG       TSLA  
Date                             
2020-01-02  28132000  142981500  
2020-01-03  23728000  266677500  
2020-01-06  34646000  151995000  
2020-01-07  30054000  268231500  
2020-01-08  30560000  467164500  
...              ...        ...  
2024-12-24   6809800   59551800  
2024-12-26   7907900   76366400  
2024-12-27  14693000   82666800  
2024-12-30  12209500   64941000  
2024-12-31  14355200   76825100  

[1258 rows x 12 columns]

do shift + tab inside the yf.download() method to explore all the parameters.
* ie. one can also pass a period as method parameter as 1d, 1m [m stands for min], 1mo [mo stands for month],10y , ytd (for year till date),etc

# 2. Getting intra-day data

## getting intraday data for 1h interval

In [6]:
#for getting the intra-day data we will make use of the interval parameter of the yf.downloadl
#as shown below

#however it is to be kept in mind that when using the interval the duration can not extends beyond 
#last 60 days
ticker="ITC.NS"
#getting 1hr data for last 1hr
yf.download(ticker, period ='1mo', interval='1h')

[*********************100%***********************]  1 of 1 completed


Price                       Adj Close       Close        High         Low  \
Ticker                         ITC.NS      ITC.NS      ITC.NS      ITC.NS   
Datetime                                                                    
2025-01-15 03:45:00+00:00  436.350006  436.350006  437.899994  435.549988   
2025-01-15 04:45:00+00:00  441.049988  441.049988  441.500000  435.850006   
2025-01-15 05:45:00+00:00  438.000000  438.000000  441.649994  437.549988   
2025-01-15 06:45:00+00:00  438.000000  438.000000  439.149994  437.850006   
2025-01-15 07:45:00+00:00  437.899994  437.899994  438.200012  437.149994   
...                               ...         ...         ...         ...   
2025-02-14 05:45:00+00:00  409.549988  409.549988  412.250000  409.000000   
2025-02-14 06:45:00+00:00  408.850006  408.850006  410.200012  408.500000   
2025-02-14 07:45:00+00:00  410.899994  410.899994  411.649994  408.750000   
2025-02-14 08:45:00+00:00  410.250000  410.250000  411.850006  409.799988   
2025-02-14 09:45:00+00:00  409.899994  409.899994  410.399994  409.500000   

Price                            Open   Volume  
Ticker                         ITC.NS   ITC.NS  
Datetime                                        
2025-01-15 03:45:00+00:00  437.899994  1388386  
2025-01-15 04:45:00+00:00  436.399994  1627845  
2025-01-15 05:45:00+00:00  441.049988  1896343  
2025-01-15 06:45:00+00:00  438.000000  1513098  
2025-01-15 07:45:00+00:00  437.950012  1188406  
...                               ...      ...  
2025-02-14 05:45:00+00:00  411.899994  1604479  
2025-02-14 06:45:00+00:00  409.600006  1497013  
2025-02-14 07:45:00+00:00  408.850006  2328067  
2025-02-14 08:45:00+00:00  410.950012  2302442  
2025-02-14 09:45:00+00:00  410.250000   726575  

[161 rows x 6 columns]

## getting intra-day data for 5min interval

In [7]:
# similarly getting 30min data for lat one month
yf.download(ticker, period='1mo', interval = '5m')

[*********************100%***********************]  1 of 1 completed


Price                       Adj Close       Close        High         Low  \
Ticker                         ITC.NS      ITC.NS      ITC.NS      ITC.NS   
Datetime                                                                    
2025-01-15 03:45:00+00:00  437.799988  437.799988  437.899994  436.149994   
2025-01-15 03:50:00+00:00  436.500000  436.500000  437.649994  436.149994   
2025-01-15 03:55:00+00:00  436.950012  436.950012  437.000000  436.149994   
2025-01-15 04:00:00+00:00  436.500000  436.500000  437.149994  436.250000   
2025-01-15 04:05:00+00:00  437.200012  437.200012  437.500000  436.700012   
...                               ...         ...         ...         ...   
2025-02-14 09:35:00+00:00  409.899994  409.899994  411.250000  409.799988   
2025-02-14 09:40:00+00:00  410.250000  410.250000  410.500000  409.950012   
2025-02-14 09:45:00+00:00  410.399994  410.399994  410.399994  409.850006   
2025-02-14 09:50:00+00:00  409.750000  409.750000  410.350006  409.500000   
2025-02-14 09:55:00+00:00  409.899994  409.899994  410.000000  409.500000   

Price                            Open  Volume  
Ticker                         ITC.NS  ITC.NS  
Datetime                                       
2025-01-15 03:45:00+00:00  437.899994  400593  
2025-01-15 03:50:00+00:00  437.500000  205537  
2025-01-15 03:55:00+00:00  436.450012   99844  
2025-01-15 04:00:00+00:00  436.950012   83576  
2025-01-15 04:05:00+00:00  436.700012   80160  
...                               ...     ...  
2025-02-14 09:35:00+00:00  411.250000  234886  
2025-02-14 09:40:00+00:00  410.000000  205142  
2025-02-14 09:45:00+00:00  410.250000  292914  
2025-02-14 09:50:00+00:00  410.350006  252295  
2025-02-14 09:55:00+00:00  409.750000  181366  

[1725 rows x 6 columns]

note that when we fetched data for 30min interval we got 161 records. however when we fetched 5min interval data we received 1725 records for the same duration i.e. 1month

## getting intra-data for 1min interval

In [8]:
# getting 1min data for last 5days
yf.download(ticker, period='5d', interval='1m')

[*********************100%***********************]  1 of 1 completed


Price                       Adj Close       Close        High         Low  \
Ticker                         ITC.NS      ITC.NS      ITC.NS      ITC.NS   
Datetime                                                                    
2025-02-10 03:45:00+00:00  429.399994  429.399994  429.899994  428.250000   
2025-02-10 03:46:00+00:00  430.450012  430.450012  430.799988  429.000000   
2025-02-10 03:47:00+00:00  430.649994  430.649994  431.299988  430.649994   
2025-02-10 03:48:00+00:00  430.899994  430.899994  430.899994  430.500000   
2025-02-10 03:49:00+00:00  430.799988  430.799988  431.200012  430.750000   
...                               ...         ...         ...         ...   
2025-02-14 09:55:00+00:00  409.850006  409.850006  409.899994  409.700012   
2025-02-14 09:56:00+00:00  409.600006  409.600006  409.799988  409.500000   
2025-02-14 09:57:00+00:00  409.899994  409.899994  409.899994  409.500000   
2025-02-14 09:58:00+00:00  409.750000  409.750000  410.000000  409.500000   
2025-02-14 09:59:00+00:00  409.899994  409.899994  409.899994  409.549988   

Price                            Open  Volume  
Ticker                         ITC.NS  ITC.NS  
Datetime                                       
2025-02-10 03:45:00+00:00  428.950012       0  
2025-02-10 03:46:00+00:00  429.500000  110030  
2025-02-10 03:47:00+00:00  430.750000   94734  
2025-02-10 03:48:00+00:00  430.600006   56580  
2025-02-10 03:49:00+00:00  430.950012   65142  
...                               ...     ...  
2025-02-14 09:55:00+00:00  409.750000   36545  
2025-02-14 09:56:00+00:00  409.700012   52578  
2025-02-14 09:57:00+00:00  409.500000   31390  
2025-02-14 09:58:00+00:00  410.000000   41008  
2025-02-14 09:59:00+00:00  409.549988   19845  

[1875 rows x 6 columns]

* however with yahoo finance we have to keep in mind that for a particualr stock if there is no activity then we will not get the data though its a valid ticker and it exists in the market for that min [when there is no transaction]
* it is also to be noted that the time that we receive in datetime is in UTC , and gives us the record from 9:15AM IST to 3:29PM IST

In [9]:
## exporting the data to csv
itc = yf.download(ticker, period='5d', interval='1m')
itc.to_csv("./data-export/exported-itc.csv")

[*********************100%***********************]  1 of 1 completed


# 3. Getting Pre-market and post-market data

In [10]:
## US market timings

# pre-market trading time
# 4:00AM EST to 9:29 am EST

# regular market trading time
# 9:30 AM EST to 4:00 PM EST

# post-market trading time
# 4.01PM EST to 8:00PM EST

In [11]:
# thus for getting the pre and post-market data we have to pass prepost=True . the default value of this parameter is False
# and hence by default the download() gives us only regular-market trading time data
ticker = "GOOG"
yf.download(ticker, period='5d', interval='1m', prepost=True)

[*********************100%***********************]  1 of 1 completed


Price                     Adj Close     Close      High     Low    Open Volume
Ticker                         GOOG      GOOG      GOOG    GOOG    GOOG   GOOG
Datetime                                                                      
2025-02-10 09:00:00+00:00  187.0200  187.0200  188.0500  187.02  187.74      0
2025-02-10 09:01:00+00:00  187.5800  187.5800  187.5800  187.08  187.09      0
2025-02-10 09:02:00+00:00  187.3100  187.3100  187.5700  187.31  187.55      0
2025-02-10 09:03:00+00:00  187.4000  187.4000  187.4100  187.15  187.33      0
2025-02-10 09:04:00+00:00  187.7000  187.7000  187.7000  187.39  187.39      0
...                             ...       ...       ...     ...     ...    ...
2025-02-15 00:55:00+00:00  187.0077  187.0077  187.0600  186.93  187.03      0
2025-02-15 00:56:00+00:00  186.9100  186.9100  186.9200  186.91  186.92      0
2025-02-15 00:57:00+00:00  186.9800  186.9800  186.9800  186.98  186.98      0
2025-02-15 00:58:00+00:00  186.9692  186.9692  186.9692  186.92  186.93      0
2025-02-15 00:59:00+00:00  186.9200  186.9200  186.9697  186.92  186.96      0

[4352 rows x 6 columns]

In [12]:
# however this is to be noted that this does not work for Indian-market and it gives us only regular-market-hour data
# for indian tickers even if we pass this parameter as True
ticker = "RELIANCE.NS"
yf.download(ticker, period='5d', interval='1m', prepost=True)

[*********************100%***********************]  1 of 1 completed


Price                        Adj Close        Close         High          Low  \
Ticker                     RELIANCE.NS  RELIANCE.NS  RELIANCE.NS  RELIANCE.NS   
Datetime                                                                        
2025-02-10 03:45:00+00:00  1259.699951  1259.699951  1263.699951  1258.050049   
2025-02-10 03:46:00+00:00  1258.449951  1258.449951  1259.900024  1258.150024   
2025-02-10 03:47:00+00:00  1255.550049  1255.550049  1258.800049  1255.550049   
2025-02-10 03:48:00+00:00  1254.800049  1254.800049  1255.449951  1253.750000   
2025-02-10 03:49:00+00:00  1255.349976  1255.349976  1255.349976  1254.500000   
...                                ...          ...          ...          ...   
2025-02-14 09:55:00+00:00  1216.500000  1216.500000  1216.550049  1216.050049   
2025-02-14 09:56:00+00:00  1216.099976  1216.099976  1216.800049  1216.000000   
2025-02-14 09:57:00+00:00  1216.050049  1216.050049  1216.199951  1216.000000   
2025-02-14 09:58:00+00:00  1215.849976  1215.849976  1216.900024  1214.050049   
2025-02-14 09:59:00+00:00  1215.300049  1215.300049  1216.000000  1214.199951   

Price                             Open      Volume  
Ticker                     RELIANCE.NS RELIANCE.NS  
Datetime                                            
2025-02-10 03:45:00+00:00  1263.699951           0  
2025-02-10 03:46:00+00:00  1259.550049       47302  
2025-02-10 03:47:00+00:00  1258.800049       36237  
2025-02-10 03:48:00+00:00  1255.150024       73731  
2025-02-10 03:49:00+00:00  1254.800049       40580  
...                                ...         ...  
2025-02-14 09:55:00+00:00  1216.050049       36478  
2025-02-14 09:56:00+00:00  1216.300049       44139  
2025-02-14 09:57:00+00:00  1216.000000       54961  
2025-02-14 09:58:00+00:00  1216.050049       69753  
2025-02-14 09:59:00+00:00  1215.400024       51962  

[1875 rows x 6 columns]

# 4. getting company info

In [13]:
ticker = "GOOG"
google = yf.Ticker(ticker)
google.info

{'address1': '1600 Amphitheatre Parkway',
 'city': 'Mountain View',
 'state': 'CA',
 'zip': '94043',
 'country': 'United States',
 'phone': '650-253-0000',
 'website': 'https://abc.xyz',
 'industry': 'Internet Content & Information',
 'industryKey': 'internet-content-information',
 'industryDisp': 'Internet Content & Information',
 'sector': 'Communication Services',
 'sectorKey': 'communication-services',
 'sectorDisp': 'Communication Services',
 'longBusinessSummary': 'Alphabet Inc. offers various products and platforms in the United States, Europe, the Middle East, Africa, the Asia-Pacific, Canada, and Latin America. It operates through Google Services, Google Cloud, and Other Bets segments. The Google Services segment provides products and services, including ads, Android, Chrome, devices, Gmail, Google Drive, Google Maps, Google Photos, Google Play, Search, and YouTube. It is also involved in the sale of apps and in-app purchases and digital content in the Google Play and YouTube;

In [14]:
# from this furthus nested attributes can be drilled down as shown below .
google.info.get('city')

'Mountain View'

In [15]:
ticker = "RELIANCE.NS"
reliance = yf.Ticker(ticker)
reliance.info.get('bookValue')

605.573

# 5.  getting dividends and stock-splits

## getting dividend and stock-split information

In [16]:
# since dividends and stock-splits are also called corporate=actions so 
#we can fetch them as shown below
ticker = "RELIANCE.NS"
reliance = yf.Ticker(ticker)
reliance.actions

Dividends  Stock Splits
Date                                              
1996-05-24 00:00:00+05:30   0.742958           0.0
1997-05-16 00:00:00+05:30   0.804871           0.0
1997-10-27 00:00:00+05:30   0.000000           2.0
1998-05-06 00:00:00+05:30   0.866784           0.0
1999-05-07 00:00:00+05:30   0.464348           0.0
2000-04-06 00:00:00+05:30   0.495305           0.0
2001-04-26 00:00:00+05:30   0.526262           0.0
2002-10-23 00:00:00+05:30   1.176349           0.0
2003-05-23 00:00:00+05:30   1.238263           0.0
2004-05-20 00:00:00+05:30   1.300176           0.0
2005-05-12 00:00:00+05:30   0.928697           0.0
2006-06-01 00:00:00+05:30   2.476525           0.0
2007-03-21 00:00:00+05:30   1.362089           0.0
2008-05-08 00:00:00+05:30   1.609741           0.0
2009-10-16 00:00:00+05:30   1.609741           0.0
2009-11-26 00:00:00+05:30   0.000000           2.0
2010-05-10 00:00:00+05:30   3.467135           0.0
2011-05-05 00:00:00+05:30   1.981220           0.0
2012-05-31 00:00:00+05:30   4.210093           0.0
2013-05-10 00:00:00+05:30   2.228873           0.0
2014-05-16 00:00:00+05:30   4.705398           0.0
2015-05-08 00:00:00+05:30   2.476525           0.0
2016-03-17 00:00:00+05:30   2.600351           0.0
2017-07-13 00:00:00+05:30   2.724178           0.0
2017-09-07 00:00:00+05:30   0.000000           2.0
2018-06-27 00:00:00+05:30   5.943660           0.0
2019-08-02 00:00:00+05:30   3.219483           0.0
2020-07-02 00:00:00+05:30   3.250000           0.0
2021-06-11 00:00:00+05:30   7.000000           0.0
2022-08-18 00:00:00+05:30   8.000000           0.0
2023-08-21 00:00:00+05:30   9.000000           0.0
2024-08-19 00:00:00+05:30   5.000000           0.0
2024-10-28 00:00:00+05:30   0.000000           2.0

## getting only dividend data

In [17]:
ticker = "RELIANCE.NS"
reliance = yf.Ticker(ticker)
reliance.dividends

Date
1996-05-24 00:00:00+05:30    0.742958
1997-05-16 00:00:00+05:30    0.804871
1998-05-06 00:00:00+05:30    0.866784
1999-05-07 00:00:00+05:30    0.464348
2000-04-06 00:00:00+05:30    0.495305
2001-04-26 00:00:00+05:30    0.526262
2002-10-23 00:00:00+05:30    1.176349
2003-05-23 00:00:00+05:30    1.238263
2004-05-20 00:00:00+05:30    1.300176
2005-05-12 00:00:00+05:30    0.928697
2006-06-01 00:00:00+05:30    2.476525
2007-03-21 00:00:00+05:30    1.362089
2008-05-08 00:00:00+05:30    1.609741
2009-10-16 00:00:00+05:30    1.609741
2010-05-10 00:00:00+05:30    3.467135
2011-05-05 00:00:00+05:30    1.981220
2012-05-31 00:00:00+05:30    4.210093
2013-05-10 00:00:00+05:30    2.228873
2014-05-16 00:00:00+05:30    4.705398
2015-05-08 00:00:00+05:30    2.476525
2016-03-17 00:00:00+05:30    2.600351
2017-07-13 00:00:00+05:30    2.724178
2018-06-27 00:00:00+05:30    5.943660
2019-08-02 00:00:00+05:30    3.219483
2020-07-02 00:00:00+05:30    3.250000
2021-06-11 00:00:00+05:30    7.000000
2022-08

## getting only stock-split data

In [18]:
ticker = "RELIANCE.NS"
reliance = yf.Ticker(ticker)
reliance.splits

Date
1997-10-27 00:00:00+05:30    2.0
2009-11-26 00:00:00+05:30    2.0
2017-09-07 00:00:00+05:30    2.0
2024-10-28 00:00:00+05:30    2.0
Name: Stock Splits, dtype: float64

## getting dividend and stock-split info as part of historical data

In [19]:
# the dividends and stock splits are also sometime referred to as corporate actions
# and in case if we want to get it has part of historical data then we have to pass 
# action=True  to the download method. the value of this parameter is false

ticker = 'GOOG'
yf.download(ticker, period='10y', actions=True)

[*********************100%***********************]  1 of 1 completed


Price        Adj Close       Close Dividends        High         Low  \
Ticker            GOOG        GOOG      GOOG        GOOG        GOOG   
Date                                                                   
2015-02-17   26.970778   27.067686       0.0   27.424706   26.980425   
2015-02-18   26.814768   26.911116       0.0   27.199823   26.801914   
2015-02-19   26.972269   27.069181       0.0   27.081148   26.826847   
2015-02-20   26.777504   26.873718       0.0   27.113062   26.716648   
2015-02-23   26.427725   26.522682       0.0   26.748512   26.398024   
...                ...         ...       ...         ...         ...   
2025-02-10  188.199997  188.199997       0.0  189.990005  187.610001   
2025-02-11  187.070007  187.070007       0.0  188.800003  186.080002   
2025-02-12  185.429993  185.429993       0.0  186.830002  183.630005   
2025-02-13  187.880005  187.880005       0.0  187.990005  184.880005   
2025-02-14  186.869995  186.869995       0.0  188.149994  186.110001   

Price             Open Stock Splits    Volume  
Ticker            GOOG         GOOG      GOOG  
Date                                           
2015-02-17   27.266640          0.0  32336537  
2015-02-18   26.995884          0.0  29061570  
2015-02-19   26.828342          0.0  19782163  
2015-02-20   27.082146          0.0  28887092  
2015-02-23   26.729115          0.0  29157834  
...                ...          ...       ...  
2025-02-10  189.059998          0.0  16606000  
2025-02-11  186.835007          0.0  13028100  
2025-02-12  185.229996          0.0  17632300  
2025-02-13  185.929993          0.0  12729300  
2025-02-14  186.830002          0.0  12673300  

[2516 rows x 8 columns]

In [20]:
#lets verify the same on some indian ticker
ticker = 'ITC.NS'
yf.download(ticker, period='10y', actions=True)

[*********************100%***********************]  1 of 1 completed


Price        Adj Close       Close Dividends        High         Low  \
Ticker          ITC.NS      ITC.NS    ITC.NS      ITC.NS      ITC.NS   
Date                                                                   
2015-02-16  193.396149  259.733337       0.0  260.666656  252.000000   
2015-02-18  195.207977  262.166656       0.0  268.000000  256.666656   
2015-02-19  193.222382  259.500000       0.0  262.466675  255.733337   
2015-02-20  196.597916  264.033325       0.0  265.266663  256.233337   
2015-02-23  193.718781  260.166656       0.0  265.866669  259.500000   
...                ...         ...       ...         ...         ...   
2025-02-10  420.561707  427.200012       0.0  433.299988  426.500000   
2025-02-11  411.799988  418.299988       0.0  430.549988  417.500000   
2025-02-12  409.899994  409.899994       6.5  413.799988  407.600006   
2025-02-13  409.700012  409.700012       0.0  413.950012  409.000000   
2025-02-14  410.250000  410.250000       0.0  416.200012  408.399994   

Price             Open Stock Splits    Volume  
Ticker          ITC.NS       ITC.NS    ITC.NS  
Date                                           
2015-02-16  252.266663          0.0  11606473  
2015-02-18  256.666656          0.0  13669321  
2015-02-19  261.966675          0.0   6980185  
2015-02-20  258.333344          0.0   8931408  
2015-02-23  263.333344          0.0   5852227  
...                ...          ...       ...  
2025-02-10  427.000000          0.0  12227140  
2025-02-11  428.799988          0.0  14288751  
2025-02-12  413.149994          0.0  13884574  
2025-02-13  409.950012          0.0  12989029  
2025-02-14  411.750000          0.0  14124236  

[2470 rows x 8 columns]

so this works equally well for indian symbols

### filtering dividends and stock-splits from historical data

In [21]:
#we can also verify this applying filter on the dataframe
ticker = 'ITC.NS'
itc = yf.download(ticker, period='10y', actions=True)
itc.columns

[*********************100%***********************]  1 of 1 completed


MultiIndex([(   'Adj Close', 'ITC.NS'),
            (       'Close', 'ITC.NS'),
            (   'Dividends', 'ITC.NS'),
            (        'High', 'ITC.NS'),
            (         'Low', 'ITC.NS'),
            (        'Open', 'ITC.NS'),
            ('Stock Splits', 'ITC.NS'),
            (      'Volume', 'ITC.NS')],
           names=['Price', 'Ticker'])

In [22]:
# filtering records for dividends
itc[itc[('Dividends', 'ITC.NS')]>0].head(5)

Price        Adj Close       Close Dividends        High         Low  \
Ticker          ITC.NS      ITC.NS    ITC.NS      ITC.NS      ITC.NS   
Date                                                                   
2015-06-03  154.126175  202.933334  4.166667  209.233337  200.399994   
2016-05-30  181.516235  237.666672  1.333333  239.000000  235.199997   
2017-06-05  246.112625  317.450012  4.750000  318.250000  313.299988   
2018-05-25  215.139923  272.299988  5.150000  275.000000  268.750000   
2019-05-22  241.364609  299.750000  5.750000  302.799988  296.799988   

Price             Open Stock Splits    Volume  
Ticker          ITC.NS       ITC.NS    ITC.NS  
Date                                           
2015-06-03  209.199997          0.0  20673919  
2016-05-30  235.600006          0.0   7362553  
2017-06-05  315.000000          0.0   9482413  
2018-05-25  272.399994          0.0   9244273  
2019-05-22  302.149994          0.0  13737694

In [23]:
# filtering to see only stock-split records
itc[itc[('Stock Splits', 'ITC.NS')]>0].head(5)

Price        Adj Close       Close Dividends    High         Low        Open  \
Ticker          ITC.NS      ITC.NS    ITC.NS  ITC.NS      ITC.NS      ITC.NS   
Date                                                                           
2016-07-01  192.730515  252.350006       0.0  259.75  242.600006  243.300003   

Price      Stock Splits    Volume  
Ticker           ITC.NS    ITC.NS  
Date                               
2016-07-01          1.5  27944232

In [24]:
#combining condition
itc[(itc[('Dividends', 'ITC.NS')]>0) | (itc[('Stock Splits', 'ITC.NS')]>0)].head(5)

Price        Adj Close       Close Dividends        High         Low  \
Ticker          ITC.NS      ITC.NS    ITC.NS      ITC.NS      ITC.NS   
Date                                                                   
2015-06-03  154.126175  202.933334  4.166667  209.233337  200.399994   
2016-05-30  181.516235  237.666672  1.333333  239.000000  235.199997   
2016-07-01  192.730515  252.350006  0.000000  259.750000  242.600006   
2017-06-05  246.112625  317.450012  4.750000  318.250000  313.299988   
2018-05-25  215.139923  272.299988  5.150000  275.000000  268.750000   

Price             Open Stock Splits    Volume  
Ticker          ITC.NS       ITC.NS    ITC.NS  
Date                                           
2015-06-03  209.199997          0.0  20673919  
2016-05-30  235.600006          0.0   7362553  
2016-07-01  243.300003          1.5  27944232  
2017-06-05  315.000000          0.0   9482413  
2018-05-25  272.399994          0.0   9244273

# 6. getting company isin

In [25]:
ticker = "RELIANCE.NS"
reliance = yf.Ticker(ticker)
reliance.isin

'INE002A01018'

# 7. getting company news

In [26]:
ticker = "RELIANCE.NS"
reliance = yf.Ticker(ticker)
reliance.news

[{'uuid': '9fe55ba6-ec08-3fca-9822-fb9685c98ffc',
  'title': 'Gold Edges Higher on Likely Technical Recovery',
  'publisher': 'The Wall Street Journal',
  'link': 'https://finance.yahoo.com/m/9fe55ba6-ec08-3fca-9822-fb9685c98ffc/gold-edges-higher-on-likely.html',
  'providerPublishTime': 1739751480,
  'type': 'STORY',
  'thumbnail': {'resolutions': [{'url': 'https://s.yimg.com/uu/api/res/1.2/Fwf5HL4GsEGPbDfR3MJBPg--~B/aD02MzA7dz0xMjAwO2FwcGlkPXl0YWNoeW9u/https://media.zenfs.com/en/wsj.com/7b2c285183be07ca3c4e8242b25a59a8',
     'width': 1200,
     'height': 630,
     'tag': 'original'},
    {'url': 'https://s.yimg.com/uu/api/res/1.2/zIJaagDuCJitt1Cvcwyc7A--~B/Zmk9ZmlsbDtoPTE0MDtweW9mZj0wO3c9MTQwO2FwcGlkPXl0YWNoeW9u/https://media.zenfs.com/en/wsj.com/7b2c285183be07ca3c4e8242b25a59a8',
     'width': 140,
     'height': 140,
     'tag': '140x140'}]}},
 {'uuid': '23da4df1-d90f-3d30-ab38-7ad68a354acd',
  'title': 'Several Insiders Invested In Noble Helium Flagging Positive News',
  'publish